batchflow/models/torch/callbacks/base.py

```diff
@@ -0,0 +1,107 @@
+""" Base class for model callbacks: describes API. """
+from abc import ABC, abstractmethod
+from time import gmtime, strftime
+
+from ....monitor import USSMonitor, GPUMonitor, GPUMemoryMonitor
+
+
+def file_print(path, msg):
+    """ Print to a file. """
+    with open(path, 'a+') as f:
+        print(msg, file=f)
+
+
+
+class BaseCallback(ABC):
+    """ Base class for callbacks. """
+    def __init__(self, stream=None):
+        self.model = None
+        self.stream = self.make_stream(stream)
+
+    def make_stream(self, stream):
+        """ Parse the `stream` argument into callable. """
+        if stream is None:
+            return lambda *args: args
+
+        if callable(stream):
+            return stream
+
+        if isinstance(stream, str):
+            with open(stream, 'w') as _:
+                pass
+            return lambda msg: file_print(stream, msg)
+        raise TypeError('`Stream` argument must be either None, callable or string.')
+
+    def set_model(self, model):
+        self.model = model
+        return self
+
+    @abstractmethod
+    def on_iter_end(self, **kwargs):
+        """ !!. """
+
+
+class LogCallback(BaseCallback):
```

Comment:
what about the tutorial in order to show how to use the callbacks?

Reply:
I actually believe that the time has come for a better `Torch` tutorial overall: a more advanced model (certainly not VGG), rich dataset (certainly not MNIST), more features from model-related like callbacks, microbatch to data augmentations. For now, I suggest this PR does the callback feature, and the demo notebook comes later